### PR TITLE
Update rank.go.tmpl

### PR DESCRIPTION
**Pull request by JordiHakimZiyech**
added an option for rank cards, if possible in the future, I will use imagekit to make the profile picture in a circle

### CODE REVIEW

File: leveling/rank.cc.lua [hunk 1]
```diff
@@ -5,13 +5,19 @@
 
 	Recommended trigger: Regex trigger with trigger `\A(-|<@!?204255221017214977>\s*)(rank|lvl|xp)(\s+|\z)`.
 */}}
+{{/* CONFIGURATION AREA STARTS */}}
+{{ $rankcard := 1 }} {{/* set to `1` for a rank card */}}
```
Comment (Jo3-L) [inline leveling/rank.cc.lua:9]: Please change this to 0 (or even better, use a boolean like a normal person) to keep the previous behavior and make this opt-in for now.

File: leveling/rank.cc.lua [hunk 2]
```diff
@@ -5,13 +5,19 @@
...
+{{/* CONFIGURATION AREA ENDS*/}}
 {{/* Instantiate constants */}}
 {{ $barEmpty := "□" }}
 {{ $barFull := "■" }}
 {{ $xp := 0 }} {{/* Xp of user */}}
 {{ $color := 14232643 }} {{/* Embed color */}} 
 {{ $user := .User }} {{/* Target user */}}
 {{ $colorSet := false }}
+{{ if (dbGet .User.ID "background") }}
+{{ $background = ((dbGet .User.ID "background").Value) }} {{end}}
```
Comment (Jo3-L) [inline leveling/rank.cc.lua:20]: Seems to be no way of setting this so far? Please remove this or add a subcommand for setting it like `set-color` is implemented currently.

Also, you can save DB call by using `with` or inlined `if` variable definition.

File: leveling/rank.cc.lua [hunk 3]
```diff
@@ -84,6 +90,11 @@
 			$embed.description
 		) }} {{/* If user in top 100, update the description */}}
 	{{ end }}
-
+{{ if $rankcard}}
+{{ if not $rank }} {{ $rank = urlescape "100" }} {{ end }}
```
Comment (Jo3-L) [inline leveling/rank.cc.lua:94]:
```suggestion
{{ $rank = or $rank "100" }}
```
`100` is already URL encoded.

File: leveling/rank.cc.lua [hunk 4]
```diff
@@ -84,6 +90,11 @@
...
+{{ $username := urlescape ($user.Username) }}
+{{ $pfp := ($user.AvatarURL "256") }}
+	{{ sendMessage nil (cembed "color" 4645612 "image" (sdict "url" (print "https://vacefron.nl/api/rankcard?username=" $username "&avatar=" $pfp "&level=" $level "&rank=" $rank "&currentxp=" $current "&nextlevelxp=" $total "&previouslevelxp=0&custombg=" $background "&xpcolor=fc1d04"))) }} {{ else }}
```
Comment (Jo3-L) [inline leveling/rank.cc.lua:97]: XP color should use the rank card color.

**Review by Jo3-L — CHANGES_REQUESTED** [review]

**Comment by Jo3-L** [comment]
Please use the `.go.tmpl` file extension as of latest commit. Thanks!

File: leveling/rank.cc.lua (hunk 1 above)
Comment (JordiHakimZiyech) [reply leveling/rank.cc.lua:9]: > Please change this to 0 (or even better, use a boolean like a normal person) to keep the previous behavior and make this opt-in for now.

but im not normmal okay :(

**Comment by JordiHakimZiyech** [comment]
whenever u see mistakes let me know, i usually make messy codes which are not good enough for the public :ye:

**Comment by Jo3-L** [comment]
There's a few merge conflicts here due to the switch we made to `.go.tmpl` (your branch is still based off the version where it was `.cc.lua`) - if you like, you can give me collab perms on your fork and I can fix it for you or open a new PR. Both are fine by me.

**Pull request closed without merging**